src/animated.tsx

```diff
@@ -189,13 +190,27 @@ export const KeyboardProvider = ({
     }
   }, [enabled]);
 
+  useEffect(() => {
+    if (__DEV__) {
+      if (
+        isEdgeToEdge() &&
+        (statusBarTranslucent !== undefined ||
+          navigationBarTranslucent !== undefined)
+      ) {
+        console.warn(
+          "statusBarTranslucent and navigationBarTranslucent props are ignored when using edge-to-edge",
+        );
+      }
+    }
+  }, [statusBarTranslucent, navigationBarTranslucent]);
+
   return (
     <KeyboardContext.Provider value={context}>
       <KeyboardControllerViewAnimated
         ref={viewTagRef}
         enabled={enabled}
-        navigationBarTranslucent={navigationBarTranslucent}
-        statusBarTranslucent={statusBarTranslucent}
+        navigationBarTranslucent={isEdgeToEdge() || navigationBarTranslucent}
```

Comment:
A small hint - let's maybe create a global variable:

```ts
const IS_EDGE_TO_EDGE = isEdgeToEdge()
```

And re-use it? Just thinking, that it can be calculated one time and re-use result. It'll be more optimal, since we will not call a function (and perform comparison) two times (or 3 times in debug).

What do you think?

Reply:
It returns a pre-computed constant, so the impect is just a function call, but I agree, for clarity it's better to update it.